### PR TITLE
Clean up via-IR equivalence command-line test

### PR DESCRIPTION
**Pull request by cameel**
A bunch of refactors extracted from #14243. No changes in functionality.

It can be reviewed and merged separately, so I want to get it out of the way.

### CODE REVIEW

File: test/cmdlineTests.sh [hunk 1]
```diff
@@ -334,16 +330,16 @@ function test_via_ir_equivalence()
...
-
-    for yul_file in $(find . -name "${output_file_prefix}*.yul" | sort -V); do
-        msg_on_error --no-stderr "$SOLC" --strict-assembly --asm "${optimizer_flags[@]}" "$yul_file" |
-            sed '/^Text representation:$/d' > "${yul_file/.yul/.asm}"
-    done
+        split_on_empty_lines_into_numbered_files "$output_file_prefix" ".yul"
 
     local asm_output_two_stage asm_output_via_ir
-    for asm_file in $(find . -name "${output_file_prefix}*.asm" | sort -V); do
-        asm_output_two_stage+=$(sed '/^asm_output_two_stage:$/d' "$asm_file" | sed '/^=======/d')
```
Comment (r0qs) [inline test/cmdlineTests.sh:346]: Why was this removed? Is it still make sense to call it `asm_output_two_stage`? Now it is only one, no?

**Review by r0qs — COMMENTED** [review]
It looks good to me, apart from one question.

File: test/cmdlineTests.sh (hunk 1 above)
Comment (cameel) [reply test/cmdlineTests.sh:346]: No, it's still two-stage. This loop was just filtering out some lines from the `--asm` output after compilation:
1) The `=======` separators. This is still happening. See the loop immediately below.
2) Lines containing `asm_output_two_stage:`. Not sure why actually since the output is not supposed to contain anything like this. Could be a left-over from how this test looked earlier. In any case now this is superfluous so I removed it.

Overall, two loops are not necessary here. It can easily be done in one, just like it's already done for `--bin` output below.

Comment (r0qs) [reply test/cmdlineTests.sh:346]: I see. Thanks

**Review by r0qs — APPROVED** [review]

File: test/cmdlineTests.sh [hunk 2]
```diff
@@ -358,16 +354,16 @@ function test_via_ir_equivalence()
 
     for yul_file in $(find . -name "${output_file_prefix}*.yul" | sort -V); do
         bin_output_two_stage+=$(
-        msg_on_error --no-stderr "$SOLC" --strict-assembly --bin "${optimizer_flags[@]}" "$yul_file" |
+            msg_on_error --no-stderr "$SOLC" --strict-assembly --bin "${optimizer_flags[@]}" "$yul_file" |
```
Comment (nikola-matic) [inline test/cmdlineTests.sh:357]: Are the newly added indents here (and the `sed`s below a mistake?

Comment (cameel) [reply test/cmdlineTests.sh:357]: No, why do you think they're a mistake?

Comment (nikola-matic) [reply test/cmdlineTests.sh:357]: Ah nvm, didn't see the open paren.
